recipes-bsp/uefi/tegra-uefi-capsules_35.3.1.bb

```diff
@@ -0,0 +1,31 @@
+DESCRIPTION = "Generate UEFI capsules for bup paylods"
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
+
+SRC_URI += "file://oe4t-set-uefi-OSIndications"
+
+inherit bup
+
+COMPATIBLE_MACHINE = "(tegra)"
+
+DEPENDS += "openssl-native tegra-flashtools-native tegra-bup-payload"
+
+L4T_DIR = "${TMPDIR}/work-shared/L4T-native-${PV}-${PR}/Linux_for_Tegra"
```

Comment:
Rather than patching the script and running it out of the shared work directory, I'd prefer moving the capsule generation logic directly into the recipe and using the `GenerateCapsule.py` tool from the EDK2 sources, as mentioned in the comments in the script.  Last I looked, the copy of that tool in the L4T kit was unmodified from the original.

We'll also need a way for the signing arguments to get filled in based on some variable settings, and a hook for calling out to a signing server.

Reply:
That makes sense.  I'll try to get started on that this week.

Reply:
Should `GenerateCapsule.py` and dependent modules be packaged into a `tegra-uefi-capsule-native` package?  I started down a different path with the patch below, but it feels 'dirty' copying over all this into `${DEPLOY_DIR_IMAGE}`.

```
diff --git a/recipes-bsp/uefi/edk2-firmware-tegra_35.3.1.bb b/recipes-bsp/uefi/edk2-firmware-tegra_35.3.1.bb
index 41ac23d2..ae6ec280 100644
--- a/recipes-bsp/uefi/edk2-firmware-tegra_35.3.1.bb
+++ b/recipes-bsp/uefi/edk2-firmware-tegra_35.3.1.bb
@@ -75,6 +75,9 @@ do_deploy() {
        [ -e ${B}/images/$dtb ] || continue
        install -m 0644 ${B}/images/$dtb ${DEPLOYDIR}/
     done
+    install -d ${DEPLOYDIR}/Common
+    install -m 0644 ${S}/BaseTools/Source/Python/Capsule/GenerateCapsule.py ${DEPLOYDIR}/
+    cp -R --preserve=mode,timestamps ${S}/BaseTools/Source/Python/Common/* ${DEPLOYDIR}/Common
 }
```

This part is still a WIP w/r/t the signing certificates, but more context on where I was headed:

```
diff --git a/recipes-bsp/uefi/tegra-uefi-capsules_35.3.1.bb b/recipes-bsp/uefi/tegra-uefi-capsules_35.3.1.bb
index c499143b..ab779f7d 100644
--- a/recipes-bsp/uefi/tegra-uefi-capsules_35.3.1.bb
+++ b/recipes-bsp/uefi/tegra-uefi-capsules_35.3.1.bb
@@ -4,17 +4,36 @@ LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda
 
 SRC_URI += "file://oe4t-set-uefi-OSIndications"
 
-inherit bup
+inherit bup l4t_bsp
 
 COMPATIBLE_MACHINE = "(tegra)"
 
-DEPENDS += "openssl-native tegra-flashtools-native tegra-bup-payload"
+DEPENDS += "openssl-native edk2-firmware-tegra tegra-bup-payload"
 
-L4T_DIR = "${TMPDIR}/work-shared/L4T-native-${PV}-${PR}/Linux_for_Tegra"
+GUID:tegra194 = "be3f5d68-7654-4ed2-838c-2a2faf901a78"
+GUID:tegra234 = "bf0d4599-20d4-414e-b2c5-3595b1cda402"
+
+def get_hex_bsp_version(bsp_version):
+    verparts = bsp_version.split('.')
+    return hex(int(verparts[0])<<16 | int(verparts[1])<<8 | int(verparts[2]))
+
+BSP_VERSION32 = "${@get_hex_bsp_version(d.getVar('L4T_VERSION'))}"
 
 do_compile() {
-    ${L4T_DIR}/generate_capsule/l4t_generate_soc_capsule.sh -i ${DEPLOY_DIR_IMAGE}/${BUPFILENAME}.bl_only.bup-payload -o ./tegra-bl.cap ${L4T_DEB_SOCNAME}
-    ${L4T_DIR}/generate_capsule/l4t_generate_soc_capsule.sh -i ${DEPLOY_DIR_IMAGE}/${BUPFILENAME}.kernel_only.bup-payload -o ./tegra-kernel.cap ${L4T_DEB_SOCNAME}
+    python3 ${DEPLOY_DIR_IMAGE}/GenerateCapsule.py \
+        -v --encode --monotonic-count 1 \
+        --fw-version "${BSP_VERSION32}" \
+        --lsv "${BSP_VERSION32}" \
+        --guid "${GUID}" \
+        -o ./tegra-bl.cap \
+        ${DEPLOY_DIR_IMAGE}/${BUPFILENAME}.bl_only.bup-payload
+    python3 ${DEPLOY_DIR_IMAGE}/GenerateCapsule.py \
+        -v --encode --monotonic-count 1 \
+        --fw-version "${BSP_VERSION32}" \
+        --lsv "${BSP_VERSION32}" \
+        --guid "${GUID}" \
+        -o ./tegra-kernel.cap \
+        ${DEPLOY_DIR_IMAGE}/${BUPFILENAME}.kernel_only.bup-payload
 }
```

Reply:
@Lexmark-chad As we discussed in the monthly meeting, I think it would make sense to create a recipes for extracting the necessary tools/scripts from EDK2.  Borrowing/adapting [this recipe](https://git.yoctoproject.org/meta-arm/tree/meta-arm/recipes-bsp/uefi/edk2-basetools-native_202302.bb?h=master) from the meta-arm layer might be enough.